andela/codepirates-ah-frontend
# #167313352 Setup react app using webpack and airbnb style quide

#### Purpose of this PR

  Setup empty react project for UI of Author's Haven

#### Tasks to be accomplished

  - Initialize project
  - Congregate requisite dependencies for project setup
  - Configure webpack package bundler for building executable code
  - Configure babel for backward code transpiling
  - Configure ESlint to set, check and enforce coding standard/style
  - Devolve environment-dependent variable management to the execution environment
  - Setup development server in webpack
  - Setup express server for production

#### How to manually test this setup

  - Clone [this](https://github.com/andela/codepirates-ah-frontend) repo
  - Create a local .env file at the root of your project directory and set values for ***host*** and ***port*** for your local development environment else, localhost:8080 defaults
  - Populate the index.js file with a piece of ES6 code, preferably involving arrow functions or other such modern javascript features
  - To **test the development and production builds,** in terminal, navigate to the root of your local app and run ***npm run dev*** and ***npm run build*** in turns, each time inspecting the contents of the created dist directory; you may or may not delete the dist directory after each run
  - To **test the development server** run ***npm run devserver*** and observe that the application opens in the browser page with title Author's Haven and the page currently displaying the logo and a welcome message. This also tests the *open* option provided to the script.
  - To **test production server** run ***npm run start*** and check that the build runs in the console and the server is started and logs Author's haven running on your local port, then in your browser's address bar, enter localhost: *your port*. You should see the logo and a welcome message displayed
  - To test the webpack config **overlay** settings, create any error, say, change an import path; like the image import path to a non existing path the in turns run the production and development servers. You should see the error logged in the browser page while in development but not when running the production server
  - To test the **hot** reload setting, run the development server and correct the error created previously in testing overlay. On correction, the page should automatically be reloaded and normal display restored. Repeat the same in production mode; you will have to manually restart the server and refresh the page to normalise after clearing the error you created 

#### Related Pivotal Tracker story

  [WIP [#167313352](https://www.pivotaltracker.com/story/show/167313352)]

#### Images

  Log of ***npm run devserver*** command

![image](https://user-images.githubusercontent.com/36167006/65474756-41ef7080-de7d-11e9-8bfb-8d86848b518a.png)
